int16 current0
int16 current1